src/libcollections/macros.rs

```diff
@@ -9,12 +9,30 @@
 // except according to those terms.
 
 /// Creates a `Vec` containing the arguments.
+///
+/// `vec!` allows `Vec`s to be defined analogous to array expressions. There
+/// are two forms of this macro:
+///
+/// - Create a `Vec` containing a given list of elements:
+///
+/// ```
+/// let v = vec![1, 2, 3];
+/// assert_eq!(v[0], 1);
+/// assert_eq!(v[1], 2);
+/// assert_eq!(v[2], 3);
+/// ```
+///
+/// - Create a `Vec` from a given element and size:
```

Comment:
I feel it should be noted that unlike the array syntax, `vec!` supports `Clone`able elements and a non-const number of elements.